yb-voyager/src/query/queryissue/parser_issue_detector.go

```diff
@@ -839,6 +895,61 @@ func (p *ParserIssueDetector) DetectMissingForeignKeyIndexes() []QueryIssue {
 	return issues
 }
 
+// DetectPrimaryKeyRecommendations recommends adding a PK when there's a UNIQUE constraint with all NOT NULL columns and no PK
+func (p *ParserIssueDetector) DetectPrimaryKeyRecommendations() []QueryIssue {
+	var issues []QueryIssue
+
+	// Skip for partitioned or inherited tables, as PK rules differ
+	shouldSkip := func(table string) bool {
+		if p.partitionedTablesMap[table] {
+			return true
+		}
+		if _, ok := p.partitionedFrom[table]; ok {
+			return true
+		}
+		if parents, ok := p.inheritedFrom[table]; ok && len(parents) > 0 {
+			return true
+		}
+		return false
+	}
+
+	for table, uLists := range p.tableUniqueConstraints {
```

Comment:
As discussed, ideally, we should be looking at unique indexes as well.

Reply:
Added a TODO and created a ticket for this too for now since the refactor is a big change:
https://yugabyte.atlassian.net/browse/DB-18078